shipyard/models/spacecraft.py

```diff
@@ -38,6 +38,8 @@ def __init__(self, hull_tonnage):
         self.bays               = list() # list of bay objects
         self.screens            = list() # list of screen objects
         self.computer           = None   # computer object
+        self.software           = list() # list of installed software
+        self.software_rating    = 0      # represents the combined rating of installed software
```

Comment:
This might be done better as derived information, rather than tracked separately. (I know we aren't consistent with this i.e. cargo is tracked and such, but I think it's the direction to start going).